tsdb/columnar.go

```diff
@@ -685,15 +603,34 @@ func (c *chunkColumnIterator) Next() (chunkenc.Chunk, error) {
 	if !ok {
 		panic("not a ByteArrayReader")
 	}
-	n, err := binaryReader.ReadByteArrays(c.chunkBuffer)
+	_, err = binaryReader.ReadByteArrays(c.chunkBuffer)
 	c.chunkBufferPos = 0
 	if err != nil && !errors.Is(err, io.EOF) {
 		return nil, err
 	}
-	fmt.Printf("Read %d chunks\n", n)
 	return c.currentChunk()
 }
 
+func (c *chunkColumnIterator) ensurePages() error {
+	// Init the pages if not already done.
+	if c.chunkPages != nil {
+		return nil
+	}
+	cols := c.pf.Root().Columns()
+	var col *parquet.Column
+	for _, c := range cols {
+		if c.Name() == "x_chunk" {
+			col = c
+			break
+		}
```

Comment:
(nit) `x_chunk` is always the fourth column from the end. Since there is an unbound number of label columns in the front, a very naive nano-optimization is to scan the columns backwards 😆